reconcile/terraform_resources.py

```diff
@@ -386,29 +390,19 @@ def run(dry_run, print_only=False,
         if disabled_deletions_detected:
             cleanup_and_exit(tf, disabled_deletions_detected)
 
-    if dry_run:
-        cleanup_and_exit(tf)
-
-    if not light:
+    if not dry_run and not light:
         err = tf.apply()
         if err:
             cleanup_and_exit(tf, err)
 
-    # Temporary skip apply secret for running tf-r per account locally.
-    # The integration running on the cluster will manage the secret
-    # after any manual running.
-    # Will refactor with caller for further operator implement.
-    if account_name:
-        cleanup_and_exit(tf)
-
-    tf.populate_desired_state(ri, oc_map, tf_namespaces)
+    tf.populate_desired_state(ri, oc_map, tf_namespaces, account_name)
 
-    actions = ob.realize_data(dry_run, oc_map, ri)
+    actions = ob.realize_data(dry_run, oc_map, ri, caller=account_name)
 
     disable_keys(dry_run, thread_pool_size,
```

Comment:
We should also add support for  aws_iam_keys to run with account_name

Reply:
very good idea! #1673

Reply:
done!